packages/views.py

```diff
@@ -0,0 +1,281 @@
+
+    name_query = request.GET.get('name')
+    street_name_query = request.GET.get('street_name')
+    street_number_query = request.GET.get('street_number')
+    city_query = request.GET.get('city')
+    number_query = request.GET.get('number')
+
+    packages = Package.objects.filter(order__partner__name=partner.name)
+    filtered_packages = Package.objects.none()
+
+    if name_query:
+        packages = packages.filter(full_name__icontains=name_query)
+        filtered_packages = packages
+    if number_query:
+        packages = packages.filter(phone_number__icontains=number_query)
+        filtered_packages = packages
+    if street_name_query:
+        packages = packages.filter(destination__street_name__icontains=street_name_query)
+        filtered_packages = packages
+    if street_number_query:
+        packages = packages.filter(destination__street_number__contains=street_number_query)
+        filtered_packages = packages
+    if city_query:
+        packages = packages.filter(destination__city__name__icontains=city_query)
+        filtered_packages = packages
+
+    orders = []
+    related_packages = []
+    for package in filtered_packages.order_by('-order__collection_date'):
+        order = package.order
+        if order not in orders:
+            orders.append(order)
+            related_packages.append([package])
+        else:
+            related_packages[orders.index(order)].append(package)
+
+    order_packages = [(orders[i], related_packages[i]) for i in range(0, len(orders))]
+    context = {
+        'partner': partner,
+        'order_packages': order_packages,
+    }
+    context.update(string_data_lists_context())
+
+    return render(request, 'packages/partner_search.html', context)
+
+
+def order_view(request, order_id):
+    order = Order.objects.get(id=order_id)
+
+    if request.POST:
+        if request.POST.get('package') == 'all':
+            for inner in order.related_packages():
+                inner.as_query().update(status=inner.next_status())
+        else:
+            package_id = request.POST.get('package')
+            package = order.related_packages().filter(id=package_id)
+            package.update(status=package.get().next_status())
+
+    packages = list(order.related_packages())
+    context = {
+        'order': order,
+        'packages': packages,
+        'is_staff': is_staff(request.user)
+    }
+
+    return render(request, "packages/order.html", context)
+
+
+def package_view(request, package_id):
+    package = Package.objects.get(id=package_id)
+    name = package.full_name.split(' ')[0] + ':'
+
+    if not name:
+        name = ''
+
+    context = {
+        'package': package,
+        'name': name
+    }
+
+    return render(request, "packages/package.html", context)
+
+
+def partner_view(request, partner_name):
+    requested_partner = Partner.objects.get(name=partner_name)
+
+    if not is_member(request.user, requested_partner.name) and not is_staff(request.user):

```

Comment:
So django has a permission system. https://docs.djangoproject.com/en/3.1/topics/auth/default/#permissions-and-authorization

Reply:
I know, but from what I've seen it didn't really align with what I wanted to do. My approach is junky, but it's quicker and easier to setup new users and partners (with potentials bugs relating to that but whatever)

Reply:
It's also inflexible, so if you every wish better permissions, you would have to deploy. The permission system is definitely capable of providing what you require here. In addition, its configurable from the _admin_ view.

I think it would be best to utilize it. 

You can assign the permission to a group, and put all the users in that group.

Reply:
opened an [issue](https://github.com/dandeduck/package-tracking-web/issues/22)

Reply:
#22